drivers/char/broadcom/vc_mem.c

```diff
@@ -99,6 +139,186 @@ vc_mem_get_current_size(void)
 }
 EXPORT_SYMBOL_GPL(vc_mem_get_current_size);
 
+static int
+vc_mem_dma_init(void)
+{
+	struct vc_mem_dma *vcdma = &vc_mem_dma;
+	struct platform_device *pdev;
+	struct device_node *fwnode;
+	struct rpi_firmware *fw;
+	struct device *dev;
+	u32 revision;
+	int rc;
+
+	if (vcdma->dev)
+		return 0;
+
+	fwnode = of_find_node_by_path("/system");
+	rc = of_property_read_u32(fwnode, "linux,revision", &revision);
+	if (((revision >> 12) & 0xf) != 2) {
+		/* Only BCM2710 has a memory hole to work around */
```

Comment:
By hole, do you mean peripherals mapping low (below 1GB) .
Doesn't that include 2709 (pi2) and 2710 (pi3)?
Wasn't pi4 still mapping low by default?

Reply:
I think you're right about 2709 - I was thinking Pi 2 was still 512MB, but that's not the case. Pi 4 is OK because the firmware puts the logs in the first 1GB while the peripherals are mapped just below 4GB, so they don't obstruct the host's view of the VPU's RAM usage.

I'll confirm that Pi 2 is affected, then add it to the conditional.